vpr/src/place/place_constraints.cpp

```diff
@@ -82,29 +82,69 @@ PartitionRegion constrained_macro_locs(const t_pl_macro& pl_macro) {
 
                 t_pl_loc min_pl_loc(reg_rect.xmin(), reg_rect.ymin(), block_regions[i].get_sub_tile());
 
-                t_pl_loc modified_min_pl_loc = min_pl_loc + offset;
+                t_pl_loc modified_min_pl_loc = min_pl_loc - offset;
 
                 t_pl_loc max_pl_loc(reg_rect.xmax(), reg_rect.ymax(), block_regions[i].get_sub_tile());
 
-                t_pl_loc modified_max_pl_loc = max_pl_loc + offset;
+                t_pl_loc modified_max_pl_loc = max_pl_loc - offset;
 
                 modified_reg.set_region_rect(modified_min_pl_loc.x, modified_min_pl_loc.y, modified_max_pl_loc.x, modified_max_pl_loc.y);
                 //check that subtile is not an invalid value before changing, otherwise it just stays -1
-                if (block_regions[i].get_sub_tile() != -1) {
+                if (block_regions[i].get_sub_tile() != NO_SUBTILE) {
                     modified_reg.set_sub_tile(modified_min_pl_loc.sub_tile);
                 }
 
                 modified_pr.add_to_part_region(modified_reg);
             }
 
             if (num_constrained_members == 1) {
-                macro_pr = modified_pr;
+                macro_head_pr = modified_pr;
             } else {
-                macro_pr = intersection(macro_pr, modified_pr);
+                macro_head_pr = intersection(macro_head_pr, modified_pr);
             }
         }
     }
 
+    //intersect to ensure the head pr does not go outside of grid dimensions
+    macro_head_pr = intersection(macro_head_pr, grid_pr);
+
+    //if the intersection is empty, no way to place macro members together, give an error
+    if (macro_head_pr.empty()) {
```

Comment:
Add detail to error message - say what blocks are in the macro that failed. Macro with x blocks can't be placed, print names of the blocks, and say that floorplan region is empty for the macro. Say to check the floorplan constraints